providers/implementations/signature/sm2_sig.c

```diff
@@ -98,6 +98,12 @@ static int sm2sig_set_mdname(PROV_SM2_CTX *psm2ctx, const char *mdname)
     if (psm2ctx->md == NULL)
         return 0;
 
+    /* XOF digests don't work */
+    if ((EVP_MD_get_flags(psm2ctx->md) & EVP_MD_FLAG_XOF) != 0) {
+        ERR_raise(ERR_LIB_PROV, PROV_R_XOF_DIGESTS_NOT_ALLOWED);
+        return 0;
+    }
```

Comment:
Does this check make sense for SM2 at all?

Reply:
I'm not aware that SM2 supports XOF digests.  I could easily be mistaken, the standards are all Chinese to me :grin:

I'm pretty sure the code doesn't support them currently.